feature/text_based.py

```diff
@@ -0,0 +1,694 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: GNU GPLv3
+
+from typing import NoReturn, Union, List, Tuple
+import random
+import pandas as pd
+import numpy as np
+from feature.base import _BaseSupervisedSelector
+from feature.utils import Num, check_true
+import matplotlib.pyplot as plt
+from mip import Model, xsum, minimize, maximize, BINARY, INTEGER, OptimizationStatus
+from sklearn.cluster import KMeans
+from scipy import sparse
+from textwiser import TextWiser
+
+
+class Data:
+    def __init__(self, cost: Union[List, np.ndarray], matrix: Union[List[List], np.ndarray]):
+        self._verify_input(cost, matrix)
+
+        # Cost Vector c[variable]
+        self.cost = cost
+
+        # Coefficient Matrix M[row][column]
+        self.matrix = matrix
+
+        # Range of variables
+        self.X = range(len(self.cost))
+
+        # Range of rows
+        self.rows = range(self.matrix.shape[0])
+
+        # Range of columns
+        self.columns = range(self.matrix.shape[1])
+
+    @staticmethod
+    def _verify_input(cost, matrix):
+        check_true(cost is not None, "Data Error: cost cannot be none")
```

Comment:
the second arguments is not text.. it needs to be an "exception". 

curious about your PyCharm setup; don't you see large marks of yellow errors coloring text here for these issues?

Reply:
Changed it with 'ValueError' to indicate that there is an issue with the values being passed to the function.